scripts/build.ps1

```diff
@@ -71,23 +72,35 @@ if (!$?) {
 }
 
 # We'll use 2x the number of cores for parallel execution
-$numberOfThreads = [int]((Get-CimInstance Win32_ComputerSystem).NumberOfLogicalProcessors) * 2
+$numberOfThreads = 2
+if ($PSVersionTable.Platform -and !$IsWindows) {
+    if ($IsMacOS) {
+        $numberOfThreads = [int](Invoke-Command -ScriptBlock { sysctl -n hw.physicalcpu }) * 2    
+    } else {
+        $numberOfThreads = [int](Invoke-Command -ScriptBlock { nproc }) * 2
+    }
+} else {
+    $numberOfThreads = [int]((Get-CimInstance Win32_ComputerSystem).NumberOfLogicalProcessors) * 2
+}
 
-$ninjaExtraTargets = ""
+$ninjaExtraTargets = @()
+if (($AppPlatform -ne "uwp") -and ($AppPlatform -ne "android")) {
+    $ninjaExtraTargets += "jsitests"
 
-if ($AppPlatform -ne "uwp") {
-    $ninjaExtraTargets += "v8windbg"
+    if (($AppPlatform -ne "linux") -and ($AppPlatform -ne "mac")) {
+        $ninjaExtraTargets += "v8windbg"
+    }
 }
 
-& ninja -v -j $numberOfThreads -C $buildoutput v8jsi jsitests $ninjaExtraTargets | Tee-Object -FilePath "$SourcesPath\build.log"
+& ninja -v -j $numberOfThreads -C $buildoutput v8jsi $ninjaExtraTargets | Tee-Object -FilePath "$SourcesPath\build.log"
 if (!$?) {
     Write-Host "Build failure, check logs for details"
     exit 1
 }
 
 Pop-Location
 
-if (!(Test-Path -Path "$buildoutput\v8jsi.dll") -and !(Test-Path -Path "$buildoutput\libv8jsi.so")) {
+if (!(Test-Path -Path "$buildoutput\v8jsi.dll") -and !(Test-Path -Path "$buildoutput\libv8jsi.so") -and !(Test-Path -Path "$buildoutput\libv8jsi.dylib")) {

```

Comment:
>libv8jsi.so [](http://example.com/codeflow?start=85&length=11)

Is it tested ? Is it linking against static CRT ?

Reply:
Not sure about the static CRT.

I ran jsitests on Linux against libv8jsi.so and they mostly all run (except for the ones involving exceptions, for some reason the exception type isn't being preserved - I didn't investigate that too deep).

Reply:
We need v8jsi linked against dynamic CRT to be consumed in RN .. so that RN and v8jsi works with the same types ..